sensors/event-handler.go

```diff
@@ -141,6 +141,30 @@ func (sec *sensorExecutionCtx) processUpdateNotification(ew *updateNotification)
 			}
 		}
 
+		// delete old status nodes if any
+	statusNodes:
+		for _, statusNode := range sec.sensor.Status.Nodes {
```

Comment:
If I understand this logic correctly, it looks it goes to infinite loop once it finds an existing node.
Could you extract this logic to a method and write simple unit tests to make sure the behavior?

Reply:
Sure.

Reply:
Added the test. Because this is a `continue` stmt instead of `goto`, it won't go into infinite loop.

Reply:
Ah, that's why! I read it wrong. but writing some tests is still good, thanks!